File: frontend/src/lib/types/course.ts
export interface Course {
    id: string;
    title: string;
    description: string | null;
    price: number;
    subject: string;
    grade: number;
    thumbnailUrl: string | null;
    isPublished: boolean;
    createdAt: string;
    updatedAt: string;
    instructorId: string;
    instructorName: string;
}

export interface CourseFilter {
    search?: string;
    subject?: string;
    grade?: number;
    minPrice?: number;
    maxPrice?: number;
    page?: number;
    pageSize?: number;
}

export interface CourseResponse {
    data: Course[];
    totalCount: number;
    page: number;
    pageSize: number;
    totalPages: number;
}

export interface CreateCourseRequest {
    title: string;
    description?: string;
    price: number;
    subject: string;
    grade: number;
}

export interface UpdateCourseRequest {
    title?: string;
    description?: string;
    price?: number;
    subject?: string;
    grade?: number;
    isPublished?: boolean;
}

File: backend/EduVN.Domain/Entities/Course.cs
namespace EduVN.Domain.Entities;

public class Course
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid InstructorId { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required string Subject { get; set; } // "Toán", "Lý", "Hóa", etc.
    public int Grade { get; set; } // 10, 11, 12
    public string? ThumbnailUrl { get; set; }
    public decimal Price { get; set; } = 0;
    public bool IsPublished { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Instructor { get; set; } = null!;
    public ICollection<Lesson> Lessons { get; set; } = new List<Lesson>();
    public ICollection<Assignment> Assignments { get; set; } = new List<Assignment>();
    public ICollection<Enrollment> Enrollments { get; set; } = new List<Enrollment>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
